pkg/operator/client/deploy_test.go

```diff
@@ -1033,3 +1038,1545 @@ version: ver2
+					},
+				},
+			},
+			want: []kotsutil.HelmChartInterface{},
+		},
+		{
+			name: "v1beta2 -- chart1 old namespace removed because namespace changed",
+			prevV1Beta2Charts: []chart{
+				{
+					name:    "chart1",
+					version: "0.0.1",
+					dirName: "chart1-release",
+				},
+				{
+					name:    "chart2",
+					version: "0.0.1",
+					dirName: "chart2-release",
+				},
+			},
+			curV1Beta2Charts: []chart{
+				{
+					name:    "chart1",
+					version: "0.0.1",
+					dirName: "chart1-release",
+				},
+				{
+					name:    "chart2",
+					version: "0.0.1",
+					dirName: "chart2-release",
+				},
+			},
+			previousV1Beta2KotsCharts: []kotsutil.HelmChartInterface{
+				&v1beta2.HelmChart{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "kots.io/v1beta2",
+						Kind:       "HelmChart",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "chart1",
+					},
+					Spec: v1beta2.HelmChartSpec{
+						Chart: v1beta2.ChartIdentifier{
+							Name:         "chart1",
+							ChartVersion: "0.0.1",
+						},
+						ReleaseName: "chart1-release",
+						Namespace:   "ns1",
+					},
+				},
+				&v1beta2.HelmChart{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "kots.io/v1beta2",
+						Kind:       "HelmChart",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "chart2",
+					},
+					Spec: v1beta2.HelmChartSpec{
+						Chart: v1beta2.ChartIdentifier{
+							Name:         "chart2",
+							ChartVersion: "0.0.1",
+						},
+						ReleaseName: "chart2-release",
+					},
+				},
+			},
+			currentV1Beta2KotsCharts: []kotsutil.HelmChartInterface{
+				&v1beta2.HelmChart{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "kots.io/v1beta2",
+						Kind:       "HelmChart",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "chart1",
+					},
+					Spec: v1beta2.HelmChartSpec{
+						Chart: v1beta2.ChartIdentifier{
+							Name:         "chart1",
+							ChartVersion: "0.0.1",
+						},
+						ReleaseName: "chart1-release",
+						Namespace:   "ns2",
+					},
+				},
+				&v1beta2.HelmChart{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "kots.io/v1beta2",
+						Kind:       "HelmChart",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "chart2",
+					},
+					Spec: v1beta2.HelmChartSpec{
+						Chart: v1beta2.ChartIdentifier{
+							Name:         "chart2",
+							ChartVersion: "0.0.1",
+						},
+						ReleaseName: "chart2-release",
+					},
+				},
+			},
+			want: []kotsutil.HelmChartInterface{
+				&v1beta2.HelmChart{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "kots.io/v1beta2",
+						Kind:       "HelmChart",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "chart1",
+					},
+					Spec: v1beta2.HelmChartSpec{
+						Chart: v1beta2.ChartIdentifier{
+							Name:         "chart1",
+							ChartVersion: "0.0.1",
+						},
+						ReleaseName: "chart1-release",
+						Namespace:   "ns1",
+					},
+				},
+			},
+		},
+		// ---- MIX OF v1beta1 and v1beta2 ---- //
```

Comment:
What do you think about adding a test case where a chart was previously v1beta1 and is now v1beta2 and we validate that it is not removed?